### PR TITLE
fix #264 mills in nighscout

### LINKED ISSUE #264: Date of treatment for "CGM Sensor Insert" or "Pump Site Change" differs with 1h in CET
In CEST everything worked fine, now in CET it is 1h to earlier calculated in UTC

### PR DESCRIPTION
**Pull request by Joerg-Schoemer**
* add a fix for #264 
* remove unneeded code
* add tests for DateExtension

### CODE REVIEW

File: nightguard.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2594,7 +2598,7 @@
 				ENABLE_BITCODE = NO;
 				EXCLUDED_ARCHS = "";
 				INFOPLIST_FILE = nightguard/Info.plist;
-				IPHONEOS_DEPLOYMENT_TARGET = 14.7;
+				IPHONEOS_DEPLOYMENT_TARGET = 16.0;
```
Comment (dhermanns) [inline nightguard.xcodeproj/project.pbxproj:2601]: Was this by accident? Or is iOS 16.0 required?

**Review by dhermanns — COMMENTED** [review]
Thanks so far! Please have a look at my question...

File: nightguard.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Joerg-Schoemer) [reply nightguard.xcodeproj/project.pbxproj:2601]: I saw that version 16 was already set in the "nightguard Widget Extension" and I needed at least version 15 for ISO8601FormatStyle, so I harmonized everything to 16

**Comment by Joerg-Schoemer** [comment]
I can turn it back to 15 if needed

File: nightguard/DateExtension.swift [hunk 2]
```diff
@@ -52,41 +52,25 @@ extension Date {
...
         
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "yyyy-MM-dd"
-        dateFormatter.locale = Locale(identifier: "en_US")
-        dateFormatter.timeZone = TimeZone.init(secondsFromGMT: 0)
-        
-        return dateFormatter.string(from: self)
+        return ISO8601DateFormatter.string(
+            from: self,
+            timeZone: TimeZone.gmt,
```
Comment (Joerg-Schoemer) [inline nightguard/DateExtension.swift:65]: this also needs 16 but can easily be changed to `TimeZone.init(identifier: "UTC")!`

Comment (dhermanns) [reply nightguard/DateExtension.swift:65]: Yes - would be good. It's always a big deal to loose people with older hardware ;) .

**Pull request merged**